docs/public-networks/how-to/bonsai-limit-trie-logs.md

```diff
@@ -0,0 +1,191 @@
+2024-02-02 05:46:11.023+00:00 | main | INFO  | TrieLogSubCommand | Estimated trie logs size after pruning: 0 B (0 B estimate is normal when using default settings)
+2024-02-02 05:46:11.023+00:00 | main | INFO  | TrieLogSubCommand | Prune ran successfully. We estimate you freed up 9 GiB!
+Prune ran successfully. We estimate you freed up 9 GiB!
+```
+
+If you are using a TOML configuration file, you can run a command similar to the following:
+
+```bash title="TOML file prune trie logs"
+sudo /usr/local/bin/besu/bin/besu --config-file=besu-config.toml storage x-trie-log prune
+```
+## Troubleshoot
+
+Troubleshoot common errors that can occur when attempting to use Bonsai Tries to reduce your database size. Ensure your command specifies the following:
+
+- `--data-storage-format`
+- `--data-path`
+- `--sync-mode`
+
+#### Prune command for mainnet
+
+The prune command should look similar to the following:
+
+```bash title="Prune command for mainnet"
+sudo /usr/local/bin/besu/bin/besu --data-path=/var/lib/besu --data-storage-format=BONSAI --sync-mode=SNAP storage x-trie-log prune
+```
+
+Ensure you stop Besu before running the command.
+
+####  Subcommand not working
+
+- Error message: `java.lang.IllegalArgumentException: Subcommand only works with data-storage-format=BONSAI`
+
+The `--data-storage-format=BONSAI` may be missing. To resolve, add the storage format. The command should look similar to the following:
+
+```bash title="P"
+sudo /usr/local/bin/besu/bin/besu --data-storage-format=BONSAI --data-path=/var/lib/besu --sync-mode=SNAP storage x-trie-log prune
+```
+
+#### Column handle not found
+
+- Error message: `java.lang.RuntimeException: Column handle not found for segment TRIE_BRANCH_STORAGE`
+
+ Ensure you specify the `data-path`. Your command should look similar to the following:
+
+```bash title="Specify data-path"
+sudo /usr/local/bin/besu/bin/besu --data-path=/var/lib/besu --data-storage-format=BONSAI --sync-mode=SNAP storage x-trie-log prune
+```
+
+#### Database not detected
+
+- Error message: `java.lang.IllegalArgumentException: Trying to retain more trie logs than chain length (0), skipping pruning`
+
+Ensure you specify the correct `--data-path` for your node. Your command should look similar to the following:
+
+```bash title="Correct data-path"
+sudo /usr/local/bin/besu/bin/besu --data-path=/var/lib/besu --data-storage-format=BONSAI --sync-mode=SNAP storage x-trie-log prune
+```
+
+#### Cannot store generated private key
+
+- Error message: `java.lang.IllegalArgumentException: Cannot store generated private key`
+
+Ensure you specify the correct `--data-path` for your node. Your command should look similar to the following:
+
+```bash title="Correct data-path"
+sudo /usr/local/bin/besu/bin/besu --data-path=/var/lib/besu --data-storage-format=BONSAI --sync-mode=SNAP storage x-trie-log prune
+```
+
+#### Valid keyPair not provided
```

Comment:
```suggestion
#### Sudo required
```

Reply:
I think the original is more findable. They might not know the resolution requires sudo

Reply:
If the intent is for a user to search for their error message, should we just change this title to match the error message?